Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/Spire/QtPromise.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+std::tuple<boost::signals2::connection,
+    QtPromise<std::vector<Nexus::OrderImbalance>>>
+    LocalOrderImbalanceIndicatorModel::subscribe(
+    const boost::posix_time::ptime& start,
+    const boost::posix_time::ptime& end,
+    const OrderImbalanceSignal::slot_type& slot) {
+  m_signals.emplace_back(OrderImbalanceSignalConnection{
+      OrderImbalanceSignal(), start, end});
+  return std::tuple<boost::signals2::connection,
+    QtPromise<std::vector<OrderImbalance>>>(
+      m_signals.back().m_imbalance_signal.connect(slot),
+      QtPromise([start, end, this] {
+        auto requested_imbalances = std::vector<OrderImbalance>();
+        for(auto& imbalance : m_imbalances) {
+          if(imbalance.m_timestamp >= start &&
+              imbalance.m_timestamp <= end) {
+            requested_imbalances.push_back(imbalance);
+          }
+        }
+        return requested_imbalances;
+    }));
+}
+
+void LocalOrderImbalanceIndicatorModel::insert(
+    const OrderImbalance& imbalance) {
+  m_imbalances.push_back(imbalance);
+  for(auto& signal : m_signals) {
```

Comment:
This is a bug unfortunately, to test this bug you will need to have a slot that calls ```subscribe```. To fix this bug you will need to iterate over ```m_signals``` using an index rather than an iterator.

Reply:
I was able to demonstrate the issue in a test, but I don't understand it. If the subscript operator also returns a reference, what's the difference?

Reply:
Yeah this is unfortunately tough to describe in a brief way, bring it up in tomorrow's meeting.